check/tests/pass.rs

```diff
@@ -871,3 +871,16 @@ type Bar = Test Foo
     let result = support::typecheck(text);
     assert!(result.is_ok(), "{}", result.unwrap_err());
 }
+
+#[test]
+fn dont_guess_a_record_when_the_construction_has_no_fields() {
+    let _ = ::env_logger::init();
+    let text = r#"
+type Test = { x : Int }
+type Test2 = Int
+
+{ Test2 }
+"#;
+    let result = support::typecheck(text);
+    assert!(result.is_ok(), "{}", result.unwrap_err());
+}
```

Comment:
Maybe `find_record` should be an exact match for the record type when it guesses a type for a record constructor.